### ISSUE TITLE
Question Page

### ISSUE THREAD

**Issue by CaptainAchilles** [hkhosrav/RiPPLE-Client-side#19]
Design a page which gives an overview of a single question.

The functionality which is encapsulated by this ticket is:
* User should be able to attempt a question
    * Have a new one recommended to them
    * Have a revision question recommended
    * Have one recommended to them to help a certain topic
* The application should use some sort of question-storing format as it needs to know what the correct answer is, and what the possible answers are

**Comment by CaptainAchilles** [comment]
Possible Schema and client side representations of the domain

``` JavaScript
{
    id: number // Question ID
    questionPayload: string // Sanitised HTML string of the question
    solution: number // The id of the possibleAnswers which is correct
    possibleAnswers: {
        id: number,
        answerPayload: string // Sanitised HTML string of the possible answer
    }[]
}
```

Something like this for Schema Storage:
``` SQL
CREATE TABLE questions (
    id int NOT NULL,
    question_payload text NOT NULL,
    PRIMARY KEY (id)
);
CREATE TABLE question_possible_answers (
    id int NOT NULL,
    for_question int NOT NULL,
    question_payload text NOT NULL,
    PRIMARY KEY (id),
    CONSTRAINT FK_for_question FOREIGN KEY (for_question) REFERENCES questions(id),
);
CREATE TABLE question_solutions (
    question_id int NOT NULL,
    answer_id int NOT NULL,
    PRIMARY KEY (question_id, answer_id),
    CONSTRAINT FK_question_solution FOREIGN KEY (question_id) REFERENCES questions(id),
    CONSTRAINT FK_question_solution_answer FOREIGN KEY (answer_id) REFERENCES question_possible_answers(id),
);
```

**Comment by hkhosrav** [comment]
Great stuff!

We could consider adding 'author' and 'date_of_creation' to the questions table.

Assuming that each possible answer is used for one particular question, which I think is a reasonable assumption, could we drop the question_solutions table?

**Comment by CaptainAchilles** [comment]
The current `Question` interface should be merged with the `Question` model discussed in #3 (which includes the author), but I agree that time-stamping is important.

I think that by dropping the `question_solutions` table and storing the correctness of the question on the `question_possible_answers` entity we needlessly lose some flexibility in the question handling. I don't think the overhead of the the lookup into the `question_solutions` table will be significant since the retrieval will be incredibly fast due it will only ever being used with primary key look-ups. 

Without a concrete plan of how we want the server/client to interact it is hard to say which option is the better to pursue, so I think programming for flexibility and organisation is the better option.
